metrics/metricshttpx/middleware.go

```diff
@@ -0,0 +1,68 @@
+package metricshttpx
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"golang.org/x/net/context"
+
+	"github.com/remind101/pkg/httpx"
+	"github.com/remind101/pkg/httpx/middleware"
+	"github.com/remind101/pkg/metrics"
+)
+
+// ResponseTimeReporter reports timing metrics using metrics package
+//
+// Usage:
+//   r := httpx.NewRouter()
+//   ...
+//   r.HandleFunc("/", func(ctx context.Context, w http.ResponseWriter, r *http.Request) error {
+//   	w.WriteHeader(http.StatusCreated)
+//   	return nil
+//   }).Methods("GET")
+//   s := NewResponseTimeReporter(r, r)
+//
+func NewResponseTimeReporter(handler httpx.Handler, router *httpx.Router) *responseTimeReporter {
+	if router == nil {
+		panic("NewResponseTimeReporter: router is requred")
+	}
+	return &responseTimeReporter{handler, router}
+}
+
+type responseTimeReporter struct {
+	handler httpx.Handler
+	router  *httpx.Router
+}
+
+func (h *responseTimeReporter) ServeHTTPContext(ctx context.Context, w http.ResponseWriter, r *http.Request) error {
```

Comment:
This is fairly normal for go actually. For me personally, it's not as big a deal since you have type information.